server/src/main/java/org/elasticsearch/search/aggregations/metrics/PercentileRanksAggregatorFactory.java

```diff
@@ -39,6 +44,22 @@
     private final PercentilesConfig percentilesConfig;
     private final boolean keyed;
 
+    static void registerAggregators(ValuesSourceRegistry valuesSourceRegistry) {
+        valuesSourceRegistry.register(PercentileRanksAggregationBuilder.NAME,
+            List.of(CoreValuesSourceType.NUMERIC, CoreValuesSourceType.HISTOGRAM),
```

Comment:
Are we intentionally not supporting dates here? Do we allow percentile ranks over date fields now?  I'm actually not sure, but we should probably make sure we're not losing functionality by accident.

Reply:
Closing the loop on this: after implementing the "supported type" test in master, and adding percentiles/ranks to the testing, we need to support `CoreValuesSourceType.NUMERIC, CoreValuesSourceType.HISTOGRAM, CoreValuesSourceType.DATE, CoreValuesSourceType.BOOLEAN`

Phew!